File: backend/instancia-3/src/main/java/br/imd/ufrn/atendimento/service/LegalImpedimentService.java
package br.imd.ufrn.atendimento.service;

import br.imd.ufrn.atendimento.dto.LegalImpedimentRequest;
import br.imd.ufrn.atendimento.dto.LegalImpedimentResponse;
import java.util.List;

public interface LegalImpedimentService {

    LegalImpedimentResponse register(LegalImpedimentRequest request);

    LegalImpedimentResponse findById(Long id);

    List<LegalImpedimentResponse> findByManifestationId(Long manifestationId);

    void remove(Long id);
}

File: frontend/instancia-3/src/app/services/legal-impediment.service.ts
import { Injectable, inject } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { LegalImpedimentRequest, LegalImpedimentResponse } from '../models/legal-impediment.model';

@Injectable({ providedIn: 'root' })
export class LegalImpedimentService {
  private readonly http = inject(HttpClient);
  private readonly base = '/api/v1/legal-impediments';

  register(request: LegalImpedimentRequest): Observable<LegalImpedimentResponse> {
    return this.http.post<LegalImpedimentResponse>(this.base, request);
  }

  findById(id: number): Observable<LegalImpedimentResponse> {
    return this.http.get<LegalImpedimentResponse>(`${this.base}/${id}`);
  }

  findByManifestationId(manifestationId: number): Observable<LegalImpedimentResponse[]> {
    return this.http.get<LegalImpedimentResponse[]>(`${this.base}/manifestation/${manifestationId}`);
  }

  remove(id: number): Observable<void> {
    return this.http.delete<void>(`${this.base}/${id}`);
  }
}
